test/tests/functions/integer.json

```diff
@@ -58,13 +58,13 @@
     },
     {
       "src": ".local $sel = {1 :integer select=exact} .local $bad = {$sel :integer} .match $bad 1 {{ONE}} * {{operand select {$bad}}}",
-      "exp": "operand select {$sel}",
+      "exp": "operand select 1",
       "expErrors": [{ "type": "bad-option" }, { "type": "bad-selector" }]
     },
     {
       "src": ".local $sel = {1 :integer select=$bad} .match $sel 1 {{ONE}} * {{variable select {$sel}}}",
       "params": [{ "name": "bad", "value": "exact" }],
-      "exp": "variable select {$sel}",
+      "exp": "variable select 1",
       "expErrors": [{ "type": "bad-option" }, { "type": "bad-selector" }]
     }
```

Comment:
shouldn't the result here be `nomatch` (we don't say what happens with `nomatch`??) and thus result in an empty pattern (and the logo)?

Reply:
No, `nomatch` is not possible with MF 2.0 because we've not reached consensus on #603. If we end up with a selector that doesn't support selection (as happens here), we format the `*` pattern.

Reply:
That section is actually really excplicit, it's just complicated because we ended up doing best-match rather than first-match selection. The bad-selector error is coming from Resolve Selectors step 2.iii.c., selecting the `*` variant is due to Filter Variants step 2.i.b.a.